### ISSUE TITLE
Tie not displayed

### ISSUE THREAD

**Issue by fablau** [opensheetmusicdisplay/opensheetmusicdisplay#1530]
Hello Simon and team,
Just encountered this problem. Look at the attached XML file, the D half-note in the right hand of the piano is tied to the next D height-note, but OSMD isn't displaying it. Any ideas why?

I have drawn in red where the tie should appear in the image below:

![TiesIssue](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/10005316/af705512-183b-4f55-bae1-f433e8fa5698)

Thanks for looking into this.

[TieIssue.xml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/14750560/TieIssue.xml.zip)

**Comment by NyuBlara** [comment]
I also encounter this issue, using the following file and OSMD@1.8.8 :

[Touront-O_gloriosa.musicxml.zip](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/files/15038825/Touront-O_gloriosa.musicxml.zip)

At measures 58/59, which, properly rendered, display like so : 

![image](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/61560065/0092e95f-8ef4-4c43-9fa1-fb5a89a7b411)

OSMD instead renders them as follows : 

![image](https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/assets/61560065/295d7d43-9cfb-435c-8303-782548d45af9)

**Comment by fablau** [comment]
@NyuBlara, I see the untied note has a dot, which makes it wrong for the defined time signature (which I guess is 4/4). Maybe in your case, that's why? Just wondering. I am sure @sschmidTU has a better explanation for both of us.

**Comment by NyuBlara** [comment]
@fablau Thank you for pointing this out : if you look at the first screenshot (taken from MuseScore 3), you will notice a tuplet notated over the dotted half note, which is missing in the second picture (taken from OSMD). 

The MusicXML does indeed specify that this is a single-note tuplet : 

`<note>
  <pitch>
	  <step>A</step>
	  <octave>4</octave>
  </pitch>
  <duration>6</duration>
  <tie type="start"/>
  <voice>1</voice>
  <type>half</type>
  <dot/>
  <time-modification>
	  <actual-notes>3</actual-notes>
	  <normal-notes>2</normal-notes>
	  <normal-type>quarter</normal-type>
  </time-modification>
  <stem>up</stem>
  <notations>
	  <tied type="start"/>
	  <tuplet type="start"/>
	  <tuplet type="stop"/>
  </notations>
</note>`

Which, if I'm not mistaken, means that this dotted half-note is actually interpreted as three quarter notes totaling the duration of one half, therefore fitting into the file's time signature (which is 2/2). 

However, OSMD does not display this single-note tuplet, which is a separate issue.

**Comment by fablau** [comment]
Oh yes, I see that now... even though it's a very unusual way of writing a tuplet. I mean, why not simply write that note as a regular single half-note? In any case, I think OSMD isn't able to handle that kind of "tuplets" (if so we can still call them!)

**Comment by fablau** [comment]
Yes, I agree!

**Comment by fablau** [comment]
@sschmidTU any thoughts about this issue?

**Comment by sschmidTU** [comment]
Fixed.
<img width="430" alt="image" src="https://github.com/user-attachments/assets/2762a2b6-06f0-4212-be30-0171dbcd9331">

This was a bug in the MusicXML file (export), not directly in OSMD, as they assigned a rest note a duration that was longer than the measure (4/4 in a 2/4 measure).
<img width="437" alt="image" src="https://github.com/user-attachments/assets/bcea203d-eaca-4a30-b2cd-09438f183c96">

You might want to tell Virtual Sheet Music Playground that they have this bug in MusicXML export (assuming every whole measure rest is 4/4), would be appreciated! We work around that case now (limiting a rest note duration to the measure duration), but it might affect other programs or issues as well.

**Comment by fablau** [comment]
Thank you Simon! Yes, you are right, we need to fix the Playground. You are right, that's a mistake from our part... I missed it! We'll rectify the problem as soon as possible.

Thank you again for looking into this!

**Comment by NyuBlara** [comment]
Thank you for the fix !

**Comment by NyuBlara** [comment]
I happen to have discovered a new sample in which a tie is missing. 

[K570-2.zip](https://github.com/user-attachments/files/17364557/K570-2.zip)

At measure 42, a Bb4 quarter note that should be prolonged by a tie has its tie missing.

Here is the note as seen in the public TS playground : 

![image](https://github.com/user-attachments/assets/6dca0e64-113d-458b-8d02-4a07859d7ad8)

And here it is in MuseScore 3.

![image](https://github.com/user-attachments/assets/bb253321-e05a-4a9b-9d86-61a4b3c7463a)

Below is the relevant portion of the MusicXML text : 

```xml
  <note default-x="14.26" default-y="-20.00" dynamics="72.22">
    <pitch>
      <step>B</step>
      <alter>-1</alter>
      <octave>4</octave>
      </pitch>
    <duration>24</duration>
    <tie type="start"/>
    <voice>1</voice>
    <type>quarter</type>
    <stem>down</stem>
    <staff>1</staff>
    <notations>
      <tied type="start"/>
      </notations>
  </note>
```

Crucially, and unlike previous cases described in this issue, the tie attribute is also missing from the resulting model notes ; this is not merely a display problem.

**Comment by sschmidTU** [comment]
@NyuBlara Fixed.
<img width="579" alt="image" src="https://github.com/user-attachments/assets/5d33e8ad-022a-4faa-b162-b2127aab25ee">

(the first/lower tie on the Bb is relevant here, the upper one on the Ab was already displayed)

We had an incorrect assumption in the code that open ties that don't exceed the measure duration can be deleted, but this Mozart sample proves that these ties can be valid.
It was a measure to combat faulty/unclosed ties in MusicXML files, but since there is no regression when removing these checks, I've removed them.

I've also simplified and added the sample to our tests:
[test_tie_missing_k570.zip](https://github.com/user-attachments/files/17397551/test_tie_missing_k570.zip)
